.travis.yml

```diff
@@ -93,6 +93,10 @@ matrix:
     #  compiler: clang
     #  os: osx
 
+    - env: HARDENING_OPTIONS="--enable-assert-fatalism"
+    - env: HARDENING_OPTIONS="--enable-assert-fatalism" SKIP_MAKE_CHECK="yes" CHUTNEY="yes"
+    - env: HARDENING_OPTIONS="--enable-assert-fatalism" SKIP_MAKE_CHECK="yes" TEST_STEM="yes"
```

Comment:
We can't add 3 extra jobs in CI. CI is already too slow for some developers.

I think our best option is to:
* add the new configure option to the global HARDENING_OPTIONS at the start of the file, and make sure all the jobs pass

If we have to use allow_failure, we can add the following jobs, without using too much more time:
* test-stem
* plain Linux gcc, no other options
But allow_failure isn't ideal anyway, because someone needs to manually check for failed jobs.

Reply:
Added `--enable-all-bugs-are-fatal` to default value of `HARDENING_OPTIONS`. For some builds I set it back to just `--enable-fragile-hardening`, because we want to keep running unit tests that check for assert failure under certain conditions. Such tests are skipped when `ALL_BUGS_ARE_FATAL` is enabled.